File: Data Structures & Algorithms/sqrtx/submission-5.py
import math

class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        res = 0
        while l <= r:
            mid = l + (r - l) // 2
            val = mid * mid
            if val == x:
                return mid
            else:
                if val > x:
                    r = mid - 1
                else:
                    l = mid + 1
                    res = mid
        return res

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var set = new HashSet<char>();
        var startOfWindow = 0;
        var endOfWindow = 0;
        var maxSubstringLength = 0;
        while (endOfWindow < s.Length) {
            if (!set.Contains(s[endOfWindow])) {
                set.Add(s[endOfWindow]);
                maxSubstringLength = Math.Max(maxSubstringLength, 1 + endOfWindow - startOfWindow);
                endOfWindow++;
            } else {
                while (set.Contains(s[endOfWindow])) {
                    set.Remove(s[startOfWindow]);
                    startOfWindow++;
                }
            }
        }

        return maxSubstringLength;
    }
}
